### PR TITLE
IEN-841 add user guide pdf to portal

### PR DESCRIPTION
**Pull request by sunghwan-park-fw**
- [x] BE - create a S3 bucket
- [x] BE - list files
- [x] BE - upload a file
- [x] BE - download a file

### CODE REVIEW

File: apps/api/src/migration/1684340041778-AddAdminAccess.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddAdminAccess1684340041778 implements MigrationInterface {
```
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply apps/api/src/migration/1684340041778-AddAdminAccess.ts:3]: At the moment, only super users have the access. If we want a new role like `admin`, we need to add an option on `Role & access control` page.

File: apps/api/src/admin/dto/upload.dto.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { IsString } from 'class-validator';
+import { ApiModelProperty } from '@nestjs/swagger/dist/decorators/api-model-property.decorator';
+
+export class UploadDTO {
+  @ApiModelProperty({
+    description: 'a user guide of pdf format',
+  })
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  file!: any;
```
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply apps/api/src/admin/dto/upload.dto.ts:9]: I think it should be `File`, but swagger crashes with it.

File: apps/api/package.json [hunk 3]
```diff
@@ -32,6 +32,7 @@
     "@nestjs/swagger": "5.1.5",
     "@nestjs/typeorm": "8.0.2",
     "@types/jsonwebtoken": ">=9.0.0",
+    "@types/multer": "^1.4.7",
```
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply apps/api/package.json]: updated

File: apps/api/src/migration/1684340041778-AddAdminAccess.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sunghwan-park-fw) [reply apps/api/src/migration/1684340041778-AddAdminAccess.ts:3]: The second query assigns the access to super admin.

File: apps/api/src/admin/dto/upload.dto.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { IsString } from 'class-validator';
+import { ApiModelProperty } from '@nestjs/swagger/dist/decorators/api-model-property.decorator';
+
+export class UploadDTO {
+  @ApiModelProperty({
+    description: 'a user guide of pdf format',
+  })
```
Comment (coryatwater-FW) [inline apps/api/src/admin/dto/upload.dto.ts:7]: think we also need `type: 'file'` to allow uploading in swagger

**Review by coryatwater-FW — APPROVED** [review]

**Pull request merged**